src/Sarif.Viewer.VisualStudio/Tags/TextViewCaretListener.cs

```diff
@@ -60,16 +84,28 @@ private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs
             UpdateAtCaretPosition(e.NewPosition);
         }
 
+        private void TextView_GotAggregateFocus(object sender, EventArgs e)
+        {
+            this.UpdateAtCaretPosition(this.textView.Caret.Position);
+        }
+
+        private void TextView_LostAggregateFocus(object sender, EventArgs e)
+        {
+            this.UpdateAtCaretPosition(this.textView.Caret.Position);
+        }
+
         private void UpdateAtCaretPosition(CaretPosition caretPosition)
         {
             SnapshotPoint caretSnapshotPoint = caretPosition.BufferPosition;
 
             NormalizedSnapshotSpanCollection normalizedSnapshotSpanCollection = new NormalizedSnapshotSpanCollection(new SnapshotSpan(start: caretSnapshotPoint, end: caretSnapshotPoint));
 
-            List<ISarifLocationTag> tagsCaretIsCurrentlyIn = this.tagger.GetTags(normalizedSnapshotSpanCollection).
+            // Handling the aggregate focus allows tags to have their highlights removed when the focus moves from
+            // one document to another rather than having a bunch of views with highlights that don't correspond
+            // with what's being selected in the solution explorer.
+            List<ISarifLocationTag> tagsCaretIsCurrentlyIn = (this.textView.HasAggregateFocus ? this.tagger.GetTags(normalizedSnapshotSpanCollection).
                 Where(tag => tag.Tag is ISarifLocationTag).
-                Select(tag => tag.Tag as ISarifLocationTag).
-                ToList();
+                Select(tag => tag.Tag as ISarifLocationTag) : Enumerable.Empty<ISarifLocationTag>()).ToList();
```

Comment:
>; [](start = 109, length = 1)

I had trouble following this because the choice of line breaks. Consider:
```
            List<ISarifLocationTag> tagsCaretIsCurrentlyIn = (this.textView.HasAggregateFocus
                ? this.tagger.GetTags(normalizedSnapshotSpanCollection)
                    Where(tag => tag.Tag is ISarifLocationTag).
                    Select(tag => tag.Tag as ISarifLocationTag)
                : Enumerable.Empty<ISarifLocationTag>()).ToList();
```